<?xml version="1.0" encoding="utf-8"?>	

<launch>

    <arg name="robot_name" default="ugv1" />
    <arg name="simulator" default="/vrep" />
    
    <arg name="num_subdivisions_phi"  default="360"/>    
    <arg name="num_subdivisions_theta"  default="180"/>        
    <arg name="max_reading"  default="50.0"/>
    <arg name="min_angle_v"  default="-10.0"/>    
    <arg name="max_angle_v"  default="45.0"/>   

    <arg name="world_frame" default="odom_$(arg robot_name)"/>
    <arg name="laser_frame" default="$(arg robot_name)/laser"/>    
    
    <arg name="cloud_in" default="$(arg simulator)/$(arg robot_name)/scan_point_cloud_color" />  <!-- from scan space-time filter -->
    <arg name="cloud_out" default="$(arg simulator)/$(arg robot_name)/expl_point_cloud" />

    <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->
      
	<node type="expl_pointcloud_filter_node" pkg="expl_pointcloud_filter" name="expl_pointcloud_filter_$(arg robot_name)" output="screen"  respawn="$(arg respawn_value)">
        <param name="laser_frame" value="$(arg laser_frame)"/>
        <param name="robot_frame" value="$(arg robot_name)/base_link"/>
        <param name="world_frame" value="$(arg world_frame)"/>

        <param name="num_subdivisions_phi"  value="$(arg num_subdivisions_phi)"/>
        <param name="num_subdivisions_theta"  value="$(arg num_subdivisions_theta)"/>        
        <param name="max_reading"  value="$(arg max_reading)"/>
        <param name="min_angle_v"  value="$(arg min_angle_v)"/>
        <param name="max_angle_v"  value="$(arg max_angle_v)"/>

        <!-- Subscribed -->      
        <remap from="point_cloud"    to="$(arg cloud_in)"/>

        <!-- Advertised -->
        <remap from="expl_point_cloud"  to="$(arg cloud_out)"/>                

	</node>
</launch>





